core-bundle/src/Image/Studio/FigureBuilder.php

```diff
@@ -0,0 +1,628 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Image\Studio;
+
+use Contao\CoreBundle\Exception\InvalidResourceException;
+use Contao\CoreBundle\File\MetaData;
+use Contao\FilesModel;
+use Contao\Image\ImageInterface;
+use Contao\Image\PictureConfiguration;
+use Contao\PageModel;
+use Contao\Validator;
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Filesystem\Filesystem;
+use Webmozart\PathUtil\Path;
+
+/**
+ * Use the `FigureBuilder` to easily create Figure result objects by applying
+ * configuration via a fluent interface. You can call `build()` multiple times
+ * (and change some settings in between) to create multiple instances.
+ */
+class FigureBuilder
+{
+    /**
+     * @readonly
+     *
+     * @var ContainerInterface
+     */
+    private $locator;
+
+    /**
+     * @readonly
+     *
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * The resource's absolute file path.
+     *
+     * @var string|null
+     */
+    private $filePath;
+
+    /**
+     * The resource's file model (if applicable).
+     *
+     * @var FilesModel|null
+     */
+    private $filesModel;
+
+    /**
+     * User defined size configuration.
+     *
+     * @phpcsSuppress SlevomatCodingStandard.Classes.UnusedPrivateElements
```

Comment:
Why is this necessary?

Reply:
In the `build()` method we're cloning the object so that the settings are frozen. The sniffer does not understand that we're accessing the clone and therefore the property *is* used. We could probably use `$this->sizeConfiguration` in L432 but this can also contain objects (e.g. `PictureConfiguration`) and therefore I think this is dangerous. 

Btw.: for consistency we imo also should use `$settings->options` in L453.

Reply:
> Btw.: for consistency we imo also should use $settings->options in L453.

I did so in cb1b4eb1373daa351b80f682d60576cf4ae961ae + also added another @phpcsSuppress annotation. It would surely be nicer if we would not need to add those, but I still prefer it over risking unstable code.